rfc/rfc-89/rfc-89.md

```diff
@@ -0,0 +1,423 @@
+to support bucket rescaling which is a replace-commit.
+
+Note: When users invoke the CALL command to modify the expression and resize historical partitions' bucket number, 
+they must follow THREE STEPS
+1. Firstly, stop all ingestion Jobs
+2. Then, trigger call command and wait for bucket rescaling completed
+3. Finally, restart all ingestion Jobs
+
+This STEPS ensures that all writers load the same expression config, maintaining consistency between the expression and the data.
+
+![workflow.jpg](workflow.jpg)
+
+Next will introduce the implementation details.
+
+### New Config
+
+| Config                                    | type   | default | Description                                                                                                                                                                                                                                                     |
+|-------------------------------------------|--------|---------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| hoodie.bucket.index.partition.rule.type   | string | regex   | Set rule parser for expressions.                                                                                                                                                                                                                                |
+| hoodie.bucket.index.partition.expressions | string | null    | Users can use this parameter to specify expression and the corresponding bucket numbers (separated by commas).Multiple rules are separated by semicolons like `hoodie.bucket.index.partition.expressions=expression1,bucket-number1;expression2,bucket-number2` |
+| hoodie.bucket.index.num.buckets           | int    | 4       | Hudi bucket number per partition.                                                                                                                                                                                                                               |
+
+The above parameters are at table-level which need to be declared in the DDL. Users can't change this config through
+runtime options like Flink /*+ OPTIONS(...) */
+
+```sql
+CREATE TABLE hudi_table(
+  id BIGINT,
+  name STRING,
+  price DOUBLE
+) WITH (
+'connector' = 'hudi',
+'path' = 'file:///tmp/hudi_table',
+'table.type' = 'MERGE_ON_READ',
+'index.type' = 'BUCKET',
+'hoodie.bucket.index.partition.expressions' = '\\d{4}-(06-(01|17|18)|11-(01|10|11)),256',
+'hoodie.bucket.index.num.buckets' = '10'
+)
+```
+
+For the dates of 06-01, 06-17, 06-18 in June and 01-11, 10-11, 11-11 in
+November of each year (in the format of yyyy-MM-dd), the corresponding bucket number for the partition is 256
+
+For common partitions use 10 as partition bucket number
```

Comment:
This is inconsistent with the default buckets number 256 and Flink default sets up as 4.

Reply:
changed.